### PR TITLE
Hot-reload support to celery

### PR DESCRIPTION
**Pull request by yotamtal**
[//]: # (Thank you for helping us out: your efforts mean great deal to the project and the community as a whole!)

[//]: # (Before you proceed:)

[//]: # (1. Make sure to add yourself to `CONTRIBUTORS.rst` through this PR provided you're contributing here for the first time)
[//]: # (2. Don't forget to update the `docs/` presuming others would benefit from a concise description of whatever that you're proposing)

## Description

This PR adds support for celery hot-reloading while developing, on changes to tasks.py and celery_app.py files when setting up a project with docker and celery.

## Rationale

[//]: # (Why does the project need that?)

I have been using this Cookiecutter template for many projects and while developing with celery tasks I had to restart the worker manually to test the changes. This minor change will allow developers to test their celery tasks faster

## Use case(s) / visualization(s)

[//]: # ("Better to see something once than to hear about it a thousand times.")

*-> Develop celery tasks -> notice a bug -> change the code -> save -> test the new code without restarting the worker manually*

### CODE REVIEW

**Comment by foarsitter** [comment]
Nice addition! 

Is the pattern not to tight? Now the restarts do not happen when there are changes made to settings.py or models.py. So why not set it to *.py?

**Comment by yotamtal** [comment]
@foarsitter I configured it that way on purpose, since I write all my tasks in those files.
I tried setting it as *.py before and then it restarts the worker on every save and the hot reload takes more time and mess up the logs, so I think this is the most efficient way to go :)

**Comment by browniebroke** [comment]
Nice addition indeed. Good question for the pattern, I'm not sure I undertsnad the problem, but I probably just need to try it out 😄

**Comment by pySilver** [comment]
Nice work! Consider using `watchgod` instead of `watchdog`.

File: {{cookiecutter.project_slug}}/compose/local/django/celery/worker/start [hunk 1]
```diff
@@ -4,4 +4,4 @@ set -o errexit
 set -o nounset
 
 
-celery -A config.celery_app worker -l INFO
+watchmedo auto-restart --directory=./ --pattern="*tasks.py;*celery_app.py" --recursive -- celery -A config.celery_app worker -l INFO
```
Comment (browniebroke) [inline {{cookiecutter.project_slug}}/compose/local/django/celery/worker/start:7]: Looks like we can do it with `watchgod` which is already installed if we use async:

```suggestion
watchgod celery.__main__.main --args -A config.celery_app beat -l INFO
```

Comment (foarsitter) [reply {{cookiecutter.project_slug}}/compose/local/django/celery/worker/start]: `beat` has to be `worker` I guess?

Comment (browniebroke) [reply {{cookiecutter.project_slug}}/compose/local/django/celery/worker/start]: Yes not sure why I suggested beat, that's definitely wrong

File: {{cookiecutter.project_slug}}/compose/local/django/celery/worker/start [hunk 2]
```diff
@@ -4,4 +4,4 @@ set -o errexit
 set -o nounset
 
 
-celery -A config.celery_app worker -l INFO
+watchgod celery.__main__.main --args -A config.celery_app beat -l INFO
```
Comment (browniebroke) [inline {{cookiecutter.project_slug}}/compose/local/django/celery/worker/start:7]:
```suggestion
watchgod celery.__main__.main --args -A config.celery_app worker -l INFO
```

**Comment by scnerd** [comment]
This is a great feature idea. Is there anything still needed before this can be merged (besides simply resolving the merge conflicts)?

**Comment by browniebroke** [comment]
Yes, I kind of forgot about it, I think it's good to go.

**Comment by browniebroke** [comment]
Thanks for your contribution @yotamtal 🎉

**Pull request merged**